common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilderTest.java

```diff
@@ -142,9 +185,35 @@ void rollingStartIntervalNumberCannotBeInFuture() {
   void failsForInvalidRollingStartIntervalNumber() {
     assertThat(
         catchThrowable(() -> DiagnosisKey.builder()
-            .withKeyData(this.expKeyData)
+            .withKeyData(expKeyData)
             .withRollingStartIntervalNumber(0)
-            .withTransmissionRiskLevel(this.expTransmissionRiskLevel).build()
+            .withTransmissionRiskLevel(expTransmissionRiskLevel).build()
+        )
+    ).isInstanceOf(InvalidDiagnosisKeyException.class);
+  }
+
+  @Test
+  void failsForInvalidOriginCountry() {
+    assertThat(
+        catchThrowable(() -> DiagnosisKey.builder()
+            .withKeyData(expKeyData)
+            .withRollingStartIntervalNumber(expRollingStartIntervalNumber)
+            .withTransmissionRiskLevel(expTransmissionRiskLevel)
+            .withCountryCode("DER")
+            .build()
+        )
+    ).isInstanceOf(InvalidDiagnosisKeyException.class);
+  }
+
+  @Test
+  void failsForInvalidVisitedCountries() {
+    assertThat(
+        catchThrowable(() -> DiagnosisKey.builder()
+            .withKeyData(expKeyData)
+            .withRollingStartIntervalNumber(expRollingStartIntervalNumber)
+            .withTransmissionRiskLevel(expTransmissionRiskLevel)
+            .withVisitedCountries(Arrays.asList("DE", "FRE"))
```

Comment:
Worth making this a parameterized test with multuiple inputs like "de" , "De" , "dE" and other combinations that are not ISO country code compliant.